dids/src/main/kotlin/web5/sdk/dids/DidMethod.kt

```diff
@@ -0,0 +1,92 @@
+package web5.sdk.dids
+
+import web5.sdk.crypto.KeyManager
+
+/**
+ * Represents options during the creation of a Decentralized Identifier (DID).
+ *
+ * Implementations of this interface may contain properties and methods that provide
+ * specific options or metadata during the DID creation processes following specific
+ * DID method specifications.
+ *
+ * ### Usage Example:
+ * Implement this interface in classes where specific creation options are needed
+ * for different DID methods.
+ *
+ * ```
+ * class MyDidCreationOptions : CreateDidOptions {
+ *     // Implementation-specific options for DID creation.
+ * }
+ * ```
+ */
+public interface CreateDidOptions
+
+/**
+ * Represents a Decentralized Identifier (DID) as per the W3C DID specification.
+ *
+ * A DID is a new type of identifier that is created, resolved, and cryptographically
+ * verified in a decentralized manner, commonly via blockchain technology.
+ *
+ * @property keyManager Reference to a [KeyManager] instance to manage cryptographic keys related to this DID.
+ * @property uri The string representation of the DID in URI format.
+ *
+ * ### Example Usage:
+ * ```
+ * val myDid = Did(myKeyManager, "did:example:123456789abcdefghi")
+ * ```
+ */
+public open class Did(
```

Comment:
This has a reference to manager, which is not nullable. I think this should be renamed so that it's clear
```suggestion
public open class DidManager(
```